File: pRobot169/python/webpost07.py
import os
import sys
import time

# python webpost07.py Chat 12/5 uid_pass.txt
gBrdName = sys.argv[1]
gDate = sys.argv[2]
file_uid_pass = sys.argv[3]

if not (gBrdName.isalnum() and gDate.count('/') == 1 and os.path.isfile(file_uid_pass) and os.path.isfile("stat.txt")):
    print(f"Usage: python {sys.argv[0]} boardname date uid_pass_file")
    print("    The form of parameter 'date' is 'mm/dd'.")
    print("    Please execute this program in a directory with 'stat.txt'.")
    sys.exit()

os.system(f"python webpost07-1.py {gBrdName} {gDate} {file_uid_pass}")
time.sleep(5)
os.system(f"python webpost07-2.py {gBrdName} {gDate} {file_uid_pass}")
time.sleep(5)
os.system(f"python webpost07-3.py {gBrdName} {gDate} {file_uid_pass}")
time.sleep(5)
os.system(f"python webpost07-4.py {gBrdName} {gDate} {file_uid_pass}")

os.remove("baha_cookie.txt")
os.remove("post_info.txt")

File: pRobot169/perl/webpost07.pl
#!/usr/bin/perl
# In version 06, all operations are put in one file. But it has
# problem when run with ActivePerl on my machine (OS: win2k SP3):
# LWP::UserAgent can only run once in one process. So I split them
# into different files, and store cookie and some information for
# posting an article.

use strict;
my $gBrdName = shift;
my $gDate = shift;
my $file_uid_pass = shift;

if (!($gBrdName =~ /^\w+$/ && $gDate =~ m#^\d\d/\d\d$#
        && -f $file_uid_pass && -f "stat.txt" )) {
    print <<USAGE;
Usage: perl $0 boardname date uid_pass_file
    The form of parameter 'date' is 'mm/dd'.
    Please execute this program in a directory with 'stat.txt'.
USAGE
    exit;
}

system("perl", "-w", "webpost07-1.pl", $gBrdName, $gDate, $file_uid_pass);
sleep 5;
system("perl", "-w", "webpost07-2.pl", $gBrdName, $gDate, $file_uid_pass);
sleep 5;
system("perl", "-w", "webpost07-3.pl", $gBrdName, $gDate, $file_uid_pass);
sleep 5;
system("perl", "-w", "webpost07-4.pl", $gBrdName, $gDate, $file_uid_pass);

unlink "baha_cookie.txt";
unlink "post_info.txt";
